extensions/shared/external-media/sources/with-media.js

```diff
@@ -53,6 +54,50 @@ export default function withMedia() {
 				};
 			}
 
+			componentDidMount() {
+				this.arrowKeysPropagationHandler = this.attachArrowKeysPropagationHandler();
+			}
+
+			componentWillUnmount() {
+				if ( this.arrowKeysPropagationHandler ) {
+					this.arrowKeysPropagationHandler.remove();
+				}
+			}
+
+			attachArrowKeysPropagationHandler = () => {
+				const eventListenerEl = document.querySelector( '.jetpack-external-media-browser' );
+				if ( ! eventListenerEl ) {
+					return null;
+				}
```

Comment:
This is dangerous, since the whole document is queried for the element, not just this component. We should instead use `ref` on the target element and add/remove listensers inside the `ref` handler.

Reply:
I wanted to use ref, but can't really see how we could do that since the Modal component doesn't allow for ref passing to the container/listener element. FYI, this won't work [when attached to the content wrapper](https://github.com/Automattic/jetpack/pull/16055/commits/ab9d7afd36ed41fbaf71ebbcc937139fe8dde701), because of [this](https://github.com/Automattic/jetpack/pull/16055#pullrequestreview-426551585). Any idea if we still can use a ref? If we can't, what would be the danger of querying that element?

Reply:
Gotcha! Since it wasn't possible to attach `ref` directly to the modal, I've tried adding a new element _within_ the modal and walk our way up though the DOM to get the modal element.

Reply:
Thanks! Just tested it locally - works like a charm!